docs/specs/clients/sign/wallet-authentication.md

```diff
@@ -0,0 +1,161 @@
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: 1
+Issued At: 2024-02-19T09:29:21.394Z
+Resources:
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+```
+
+The recap in the message
+
+```
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+```
+
+after removing the prefix (`urn:recap:`) and decoding would result in
+
+```
+{
+   "att":{
+      "eip155":{
+         "request/eth_signTypedData_v4":[{}],
+         "request/personal_sign":[{}]
+      }
+   }
+}
+```
+
+This ReCap Object is generated by the requesting application and is used to build the statement
+
+```
+I further authorize the stated URI to perform the following actions on my behalf: (1) 'request': 'eth_signTypedData_v4', 'personal_sign' for 'eip155'.
+```
+
+Any data following above format & [ERC-5573 specs](https://eips.ethereum.org/EIPS/eip-5573) can be encoded and requested as part of Authentication request, even multiple recaps.
+
+```
+http://example.com wants you to sign in with your Ethereum account:
+0x3613699A6c5D8BC97a08805876c8005543125F09
+
+I further authorize the stated URI to perform the following actions on my behalf: (1) 'request': 'eth_signTypedData_v4', 'personal_sign' for 'eip155'. (2) 'push': 'messages', 'notification' for 'eip155'. (3) 'receive': 'messages', 'notification' for 'eip155'.
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: 1
+Issued At: 2024-02-19T09:29:21.394Z
+Resources:
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+- https://example.com
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InB1c2gvbWVzc2FnZXMiOlt7fV0sInB1c2gvbm90aWZpY2F0aW9uIjpbe31dfX19
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlY2VpdmUvbWVzc2FnZXMiOlt7fV0sInJlY2VpdmUvbm90aWZpY2F0aW9uIjpbe31dfX19
+```
+
+### Authenticated sessions
+
+As mentioned above, applications utilize ReCaps to request consent from users to establish authenticated session to be able to send additional requests for the approved methods & chains to the users' wallets
+
+Each authentication request includes an array of requested `chains`. Users can choose to approve all or part of the chains depending on their wallet support.
+If only subset of the requested chains is supported by the wallet e.g.
+
+```
+requested: ['eip155:1', 'eip155:2']
+supported: ['eip155:1']
+```
+
+Only `eip155:1` is to be approved for the session. Then, the recap containing the methods requested is to be updated as follows
+
+```
+{
+   "att":{
+      "eip155":{
+         "request/eth_signTypedData_v4":[{
+            chains: ['eip155:1']
+         }],
+         "request/personal_sign":[{
+            chains: ['eip155:1']
+         }]
+      }
+   }
+}
+```
```

Comment:
I think we need to add another example here where two chains are supported and add some language around the authentication

Consider the following scenarios when chains `eip155:1` and `eip155:2` are `requested`

Scenario A - the wallet only supports `eip155:1` then we simply receive a one CACAO with an array that includes only `eip155:1`  (This is already included in the example above)

Scenario B - the wallet supports both `eip155:1  and `eip155:2` then we receive two CACAOs with an array that includes both `eip155:1` and `eip155:2` in each CACAO (this means that we can guarantee the wallet is authenticated for both chains and we can send requests for both chains)

Scenario C -  the wallet supports both `eip155:1  and `eip155:2` BUT we only receive one CACAO with an array that includes both `eip155:1` and `eip155:2` (this means that we CANNOT guarantee the wallet is authenticated for both chains BUT we can send requests for both chains)

Reply:
^ Totally agree, this would clarify the difference between what a SIWE signature "proves" and what a ReCap "expresses" (and can maybe take on faith when combined with the SIWE signature), since they are both authorization statements.

Reply:
Extra examples added 👍